tests/utils_tests/test_dateparse.py

```diff
@@ -25,6 +25,10 @@ def test_parse_time(self):
         self.assertEqual(parse_time('10:20:30.400'), time(10, 20, 30, 400000))
         self.assertEqual(parse_time('10:20:30,400'), time(10, 20, 30, 400000))
         self.assertEqual(parse_time('4:8:16'), time(4, 8, 16))
+        # Valid inputs due to optional regex capture groups
+        self.assertEqual(parse_time('00:05: '), time(0, 5))
```

Comment:
I think perhaps the regular expression should be tightened up instead as this isn't really a valid time string.

Reply:
I agree it _should_ be (it was entirely luck that I tried it), but it's a separate body of work (ie: new ticket) because it could affect inputs in the wild I think -- parse_time is used by `models.TimeField`.
By adding it in here, I can document & demonstrate that it still works (because it's falling back to the regex variant), and then the adjustment of `time_re` can subsequently cause the test to fail, at which point it can be removed or changed to an `assertRaises` to demonstrate it's no longer acceptable.

Also I don't really want to touch `time_re` myself right now ;)

Reply:
Marking as resolved, I've opened a ticket (https://code.djangoproject.com/ticket/32904) to track what might want to be done about these edge cases.

Reply:
Thanks. It looks as though `time_re` isn't anchored with `$` so it can match any string with arbitrary stuff at the end.